jenkins/hourly_maintenance.py

```diff
@@ -194,6 +196,29 @@ def DeleteCorruptGitRepos():
     return err
 
 
+def DeleteOldWorkspaces():
+    err = 0
+    live_jobs = None
+    for host in ('jenkins-master', 'pull-jenkins-master'):
+        try:
+            resp = requests.get("http://%s:8080/api/json?pretty=true&tree=jobs[name]" % host)
+            resp.raise_for_status()
+            live_jobs = {job['name'] for job in resp.json()['jobs']}
+        except requests.exceptions.ConnectionError:
+            continue
+        except requests.exceptions.RequestException:
+            traceback.print_exc()
+    if live_jobs is None:
+        print 'unable to determine live jenkins jobs, not deleting any workspaces'
+        return 1
+    for dirname in sorted(glob.glob('/var/lib/jenkins/workspace/*')):
+        key = os.path.basename(dirname).replace('@tmp', '')
+        if key not in live_jobs:
```

Comment:
I guess the assumption here is that anything that isn't constantly running we want to delete? I'm worried this might make us vulnerable to race conditions.